### ISSUE TITLE
Issue with auto generating an iOS cert

### ISSUE THREAD

**Issue by cfjedimaster** [NativeScript/sidekick-feedback#283]
### Did you verify whether this issue has already been reported here?
Yes
### Tell us about the problem
I'm trying to use the autogenerate feature to generate an iOS cert. I initially had issues because my developer account was locked. I unlocked and confirmed I could login to the Apple Developer portal. I returned to Sidekick and tried to generate again and got this:

![image](https://user-images.githubusercontent.com/393660/46973084-de15d280-d085-11e8-8408-0d3e90b108e6.png)

### Which platform(s) does the issue occur on?
iOS
Windows 10

### Provide the following version numbers that the issue occurs with:

NativeScript Sidekick Desktop
Version: 1.13.0-v.2018.10.5.2 (latest)
NativeScript CLI version: 4.2.4
CLI extension nativescript-cloud version: 1.14.0
CLI extension nativescript-starter-kits version: 0.3.5

### Tell us how to recreate the issue in as much detail as possible
1. I described it as best I can above.

### Does this issue happen every time?
Yes

### Send the client logs
https://gist.github.com/cfjedimaster/9b129a15f105a33a1491b9f64effb5da

**Comment by ggarabedian** [comment]
@cfjedimaster, the issue is occurring because the `org.nativescript.app4` bundle id (Application Identifier) is already in use. This is the unique identifier of the app and if someone has generated code signing assets with it, you cannot generate yours with the same value. With this in mind, if you change the application identifier to `org.nativescript.app467890`, the temporary certificate and mobile provisions should be generated successfully.

The error message is not very descriptive as to why the issue is occurring, unfortunately, it originates from the service (Fastlane) we use to generate the codesigning assets and we don't have control over it.

**Comment by ggarabedian** [comment]
While we cannot add this information directly in the UI as it won't look very cool, we plan to change the "Need help" link to "Troubleshooting" and point it towards an article in the documentation that will contain a list of the most common issues with the Code Signing Assistance. What do you think about this approach?

**Comment by cfjedimaster** [comment]
I think I'd care more about quick help then it being cool. ;) I'm a super new user but my worry is that _this_ particular issue could hit a lot of people. 

What if in the New App UI you add some quick text about this issue? Look at the white space available:

![image](https://user-images.githubusercontent.com/393660/47297149-e87e2200-d5d9-11e8-8001-c9e6183e1378.png)

**Comment by ggarabedian** [comment]
@cfjedimaster, I completely agree that the helpfulness of the UI shouldn't be neglected at the expense of its coolness. Nevertheless, the modern UI design relies heavily on minimalism and large chunks of texts are not very welcome. :) 

As for your suggestion to add this in the `Create App` page, I am not sure how we can fit information specific to the Code Signing Assistance there. Showing unrelated error messages in a completely different view will look strange and might confuse people more than help them. The only option that comes to my mind is to specify that the App ID must be unique in order to use the Code Signing Assistance feature, but I am not sure of how much help this would be as well.

**Comment by cfjedimaster** [comment]
I didn't mean to show the error per se, something along the lines of:

"When specifying your App ID, note that iOS requires *globally* unique IDs. Avoid using an ID that may be in use already, like org.nativescript.test."

**Comment by ggarabedian** [comment]
Thank you for the clarification! This makes sense, especially considering that to publish an app to Google Play it needs to have a globally unique App ID as well. I will run this through our development and UX teams and unless they have any concerns, we will update the message accordingly.
